scripts/networking/network-verify/script.sh

```diff
@@ -0,0 +1,215 @@
+#!/bin/bash
+#
+# Description: Downloads latest osdctl binary and runs network egress verification in pod mode.
+
+set -euo pipefail
+
+# Global variables
+readonly NAMESPACE="openshift-network-diagnostics"
+readonly OSDCTL_REPO="openshift/osdctl"
+readonly OSDCTL_BINARY="/tmp/osdctl"
+readonly LOG_INFO="INFO:"
+readonly LOG_ERROR="ERROR:"
+readonly LOG_WARN="WARNING:"
+
+# Validate environment for Kubernetes access
+validate_environment() {
+    # Check if running in a pod with service account
+    if [[ -f "/var/run/secrets/kubernetes.io/serviceaccount/token" ]]; then
+        echo "$LOG_INFO Running in pod with ServiceAccount token"
+        return 0
+    fi
+    
+    # Check if oc is available and logged in
+    if command -v oc &> /dev/null; then
+        if oc whoami &> /dev/null; then
+            echo "$LOG_INFO Logged into OpenShift cluster via oc"
+            return 0
+        fi
+    fi
+    
+    echo "$LOG_ERROR Must be running in a pod with ServiceAccount token or logged into OpenShift cluster"
+    return 1
+}
+
+# Detect the cloud provider, region, and platform from cluster metadata
+detect_platform_and_region() {
+    local region=""
+    local cloud_provider=""
+    local platform=""
+
+    # Try to get cloud provider from cluster infrastructure
+    cloud_provider=$(oc get infrastructure cluster -o jsonpath='{.status.platform}' 2>/dev/null || echo "")
+    
+    if [[ -z "$cloud_provider" ]]; then
+        echo "$LOG_ERROR Could not determine cloud provider from cluster infrastructure" >&2
+        return 1
+    fi
+    
+    case "$cloud_provider" in
+        "AWS")
+            region=$(oc get infrastructure cluster -o jsonpath='{.status.platformStatus.aws.region}' 2>/dev/null || echo "")
+            platform="aws-classic"
+            if [[ -n "$region" ]]; then
+                echo "$region $platform"
+                return 0
+            else
+                echo "$LOG_ERROR Could not determine AWS region" >&2
+                return 1
+            fi
+            ;;
+        "GCP")
+            region=$(oc get infrastructure cluster -o jsonpath='{.status.platformStatus.gcp.region}' 2>/dev/null || echo "")
```

Comment:
Region is not a required flag in `osdctl` for network verification. It's _only_ required for AWS https://github.com/openshift/osdctl/blob/c671e124293051c83c3ea0f50a20af60b5bc2130/cmd/network/verification.go#L574-L581